### PR TITLE
manifest: hal_nordic: Update nrf_802154 driver

### PR DESCRIPTION
**Pull request by ankuns**
This commit brings latest changes to the nRF 802.15.4 Radio Driver.

### CODE REVIEW

**Comment by ankuns** [comment]
@rlubos @carlescufi , it seems that no bot was able to add reviewers automatically but I cannot also request manually for review.

**Review by rlubos — APPROVED** [review]

**Comment by ankuns** [comment]
Bump, please merge if state of release allows.

**Comment by carlescufi** [comment]
Waiting for an approval from @anangl

**Review by anangl — APPROVED** [review]

**Pull request merged**
